lib/ansible/galaxy/collection.py

```diff
@@ -0,0 +1,923 @@
+                manifest_entry['chksum_sha256'] = secure_hash(abs_path, hash_func=sha256)
+
+                manifest['files'].append(manifest_entry)
+
+    _walk(collection_path, collection_path)
+
+    return manifest
+
+
+def _build_manifest(namespace, name, version, authors, readme, tags, description, license_ids, license_file,
+                    dependencies, repository, documentation, homepage, issues, **kwargs):
+
+    manifest = {
+        'collection_info': {
+            'namespace': namespace,
+            'name': name,
+            'version': version,
+            'authors': authors,
+            'readme': readme,
+            'tags': tags,
+            'description': description,
+            'license': license_ids,
+            'license_file': license_file,
+            'dependencies': dependencies,
+            'repository': repository,
+            'documentation': documentation,
+            'homepage': homepage,
+            'issues': issues,
+        },
+        'file_manifest_file': {
+            'name': 'FILES.json',
+            'ftype': 'file',
+            'chksum_type': 'sha256',
+            'chksum_sha256': None,  # Filled out in _build_collection_tar
+            'format': MANIFEST_FORMAT
+        },
+        'format': MANIFEST_FORMAT,
+    }
+
+    return manifest
+
+
+def _build_collection_tar(collection_path, tar_path, collection_manifest, file_manifest):
+    files_manifest_json = to_bytes(json.dumps(file_manifest, indent=True), errors='surrogate_or_strict')
+    collection_manifest['file_manifest_file']['chksum_sha256'] = secure_hash_s(files_manifest_json, hash_func=sha256)
+    collection_manifest_json = to_bytes(json.dumps(collection_manifest, indent=True), errors='surrogate_or_strict')
+
+    with _tempdir() as temp_path:
+        tar_filepath = os.path.join(temp_path, os.path.basename(tar_path))
+
+        with tarfile.open(tar_filepath, mode='w:gz') as tar_file:
+            # Add the MANIFEST.json and FILES.json file to the archive
+            for name, b in [('MANIFEST.json', collection_manifest_json), ('FILES.json', files_manifest_json)]:
+                b_io = BytesIO(b)
+                tar_info = tarfile.TarInfo(name)
+                tar_info.size = len(b)
+                tar_info.mtime = time.time()
+                tar_info.mode = 0o0644
+                tar_file.addfile(tarinfo=tar_info, fileobj=b_io)
+
+            for file_info in file_manifest['files']:
+                if file_info['name'] == '.':
+                    continue
+
+                filename = to_text(file_info['name'], errors='surrogate_or_strict')
+                src_path = os.path.join(to_text(collection_path, errors='surrogate_or_strict'), filename)
+
+                if os.path.islink(src_path) and os.path.isfile(src_path):
+                    src_path = os.path.realpath(src_path)
+
+                def reset_stat(tarinfo):
+                    if tarinfo.issym() or tarinfo.islnk():
+                        return None
```

Comment:
```suggestion
                        return False
```
?

Reply:
Why are you suggesting this? Even so after testing it again it does not seem to be needed as we run `os.path.realpath` so no links should be detected here.